### PR TITLE
feat: chan-less yoinking

### PR DESCRIPTION
**Pull request by brian6932**

### CODE REVIEW

**Review by melon095 — CHANGES_REQUESTED** [review]
This won't work in the current state.
For the bot to be able to modify the recipients emote-set, the bot first has to know that users emote-set at the moment it would mutate the current channels emote-set.

You would have to figure out the recipients users emote-set and have the `addEmote` function take in that data, rather than the `EmoteSet` function.

`getSevenTVAccount` should do that for you.

**Review by brian6932 — COMMENTED** [review]
The reason this was done was because ECMAScript regex look-behinds are fixed width, and there is no `\K`, so I wouldn't be able to extract and ignore both `#` and `@` chars with a single regex.
The other option was running a regex on each index, or checking first char of each index, the latter seemed preferable.

**Comment by brian6932** [comment]
Since multiple chans don't really make sense being filtered, I think potentially a better way to do this would be getting the first index of a chan, then removing that in a reduce step, leaving all the other indexes for the possibilities of them being emotes as well 🤔

File: src/commands/yoink.ts [hunk 1]
```diff
@@ -106,7 +125,7 @@ export default class extends CommandModel<PreHandlers> {
 	};
 	LongDescription = async (prefix: string) => [
 		'Steal several 7TV emotes from a channel.',
-		'Requires that you have editor status in the current channel.',
+		'If the current channel is not specified, target will be set to the current channel, and the bot will add to the user\'s channel',
```
Comment (melon095) [inline src/commands/yoink.ts]: Nitpicket. Could add an example here if there no channel is specified

File: src/commands/yoink.ts [hunk 2]
```diff
@@ -37,20 +37,31 @@ export default class extends CommandModel<PreHandlers> {
...
+				return emotes
+
+			if (!caseSensitive)
+				emote = emote.toLowerCase()
+
+			emotes.push(emote)
+			return emotes
+		}, [])
+
+		let srcUser: any;
```
Comment (melon095) [inline src/commands/yoink.ts]: We can't use `any` here.

**Review by melon095 — CHANGES_REQUESTED** [review]

File: src/commands/yoink.ts [hunk 3]
```diff
@@ -85,7 +90,15 @@ export default class extends CommandModel<PreHandlers> {
 
 		const promises: Promise<[string, ChangeEmoteInset]>[] = [];
 
-		toAdd.forEach((emote) => promises.push(addEmote(emote, EmoteSet(), keepAlias)));
+		const emoteset: string | undefined = (await gql.GetUser(await Bot.User.ResolveUsername(srcUser.username))).connections.find(i => i.platform === ConnectionPlatform.TWITCH)?.emote_set_id
```
Comment (melon095) [inline src/commands/yoink.ts]: unnecessary call to 7tv. We already have the user fetched `:58`. And im pretty sure the code above prevents `:95` from being required.

**Review by melon095 — CHANGES_REQUESTED** [review]

**Comment by melon095** [comment]
plink

Thank you for helping me. I very much appreciate it.

**Review by melon095 — APPROVED** [review]

**Pull request merged**